src/infrastructure/figma/transformers/figma-node-transformer.ts

```diff
@@ -60,18 +64,39 @@ export const tryTransformNodeToAtlassianDesign = ({
 	const { node, extra } = nodeData;
 	const fileName = fileResponse.name;
 
+	const prevDevStatus = devStatus ?? AtlassianDesignStatus.NONE;
+
+	const nodeDevStatus = extra.devStatus
+		? mapNodeStatusToDevStatus(extra.devStatus)
+		: AtlassianDesignStatus.NONE;
+
+	let nodeLastModified = extra.lastModified;
+	if (nodeDevStatus !== prevDevStatus) {
+		// Always treat the devStatus we just received as the source of truth
+		// so update the lastModifiedTime to be that of the file since devStatus
+		// is not tracked at a node level
+		nodeLastModified = fileResponse.lastModified;
+	} else if (devStatusLastModified != null) {
+		// If the timestamp we have from changing the dev status is more recent than
+		// the node's last modified time, we should use that value instead.
+		if (
+			getUpdateSequenceNumberFrom(devStatusLastModified) >
+			getUpdateSequenceNumberFrom(nodeLastModified)
+		) {
+			nodeLastModified = devStatusLastModified;
```

Comment:
How can we be sure this won't cause a race condition where the `devStatusLastModified` and `fileResponse.lastModified` collide or are processed out of order?